### ISSUE TITLE
Attention dropout

### ISSUE THREAD

**Issue by entrpn** [jenkspt/gpt-jax#1]
Hello,

Was looking at your code and noticed that in the [Attention module](https://github.com/jenkspt/gpt-jax/blob/e22564f0778a4d0ac36436a78fdb100fef2cb98f/model.py#L47), you're doing dropout and setting it to x, but on the next line, you're multiplying your att w/ v and setting that to x so dropout doesn't actually get applied. Am I reading that correctly?

```python
x = nn.Dropout(self.dropout_rate)(attn, deterministic=deterministic)

# return weighted sum over values for each query position
x = jnp.einsum('...hqk,...khd->...qhd', attn, v).reshape(B, T, C)
x = nn.Dense(C, dtype=self.dtype, name='c_proj')(x)
```

Thanks!

**Comment by jenkspt** [comment]
you're correct -- good catch! Missed this in my tests since dropout is turned off. Just pushed the fix
